master/lustrefactory.py

```diff
@@ -72,6 +83,54 @@ def buildCommand(props):
 
     return args
 
+def getTarballFactory(gerrit_repo, **kwargs):
+    """ Generates a build factory for a tarball generating builder.
+    Args:
+        gerrit_repo (string): Gerrit repo url
+    Returns:
+        BuildFactory: Build factory with steps for generating tarballs.
+    """
+    bf = util.BuildFactory()
+
+    # Pull the patch from Gerrit
+    bf.addStep(Gerrit(
+        repourl=gerrit_repo, workdir="build/lustre",
+        mode="full", method="fresh", retry=[60,60], timeout=3600,
+        logEnviron=False, getDescription=True, haltOnFailure=True,
+        description=["cloning"], descriptionDone=["cloned"]))
+
+    # make tarball
+    bf.addStep(ShellCommand(
+        command=['sh', './autogen.sh'],
+        description=["autogen"],
+        descriptionDone=["autogen"],
+        workdir="build/lustre"))
+
+    bf.addStep(Configure(
+        command=['./configure', '--enable-dist'],
+        workdir="build/lustre"))
+
+    bf.addStep(Compile(
```

Comment:
I'm changing that step to a ShellCommand. The Compile build step object does some fancy warning count stuff, but it's not really useful to us.